zulipterminal/ui_tools/views.py

```diff
@@ -619,6 +621,7 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             self.set_focus('header')
             return key
         elif is_command_key('GO_BACK', key):
+            self.user_search.reset_search_text()
             self.allow_update_user_list = True
             self.body = UsersView(self.users_btn_list)
             self.set_body(self.body)
```

Comment:
The above two lines aren't your change, but looking at your diff I'm wondering if we need to create a new `UsersView` each time, compared to how it is done in the other methods.